### PR TITLE
chore: add new sdk metrics event types to proto

### PR DESCRIPTION
**Pull request by ono-max**

### CODE REVIEW

**Review by masaaania — APPROVED** [review]
LGTM! 👍

File: proto/event/client/event.proto [hunk 1]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
...
 message TimeoutErrorCountMetricsEvent {
   string tag = 1;
 }
 
+// TODO: Remove this event after grpc server is removed.
 message InternalErrorCountMetricsEvent {
   string tag = 1;
 }
 
+message TimeoutErrorNumberMetricsEvent {
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:113]: nit: `TimeoutErrorMetricsEvent`

File: proto/event/client/event.proto [hunk 2]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
...
+  APIId api_id = 1;
+  string tag = 2 [deprecated = true];
+  map<string, string> labels = 3;
+}
+
+message InternalErrorNumberMetricsEvent {
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:119]: nit: `InternalServerErrorMetricsEvent`

File: proto/event/client/event.proto [hunk 3]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
...
 }
 
+// TODO: Remove this event after grpc server is removed.
 message InternalErrorCountMetricsEvent {
   string tag = 1;
 }
 
+message TimeoutErrorNumberMetricsEvent {
+  APIId api_id = 1;
+  string tag = 2 [deprecated = true];
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:115]: If we keep the `TimeoutErrorCountMetricsEvent` until it is removed, why do we need to add a deprecated property in a new proto message?

File: proto/event/client/event.proto [hunk 4]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
   google.protobuf.Any event = 2;
 }
 
+// TODO: Remove this event after grpc server is removed.
 message GetEvaluationLatencyMetricsEvent {
   map<string, string> labels = 1;
   google.protobuf.Duration duration = 2;
 }
 
+// TODO: Remove this event after grpc server is removed.
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:78]: Same as above.

File: proto/event/client/event.proto [hunk 5]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
   google.protobuf.Any event = 2;
 }
 
+// TODO: Remove this event after grpc server is removed.
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:72]: Please make it deprecated instead of writing a comment.

File: proto/event/client/event.proto [hunk 6]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
...
   google.protobuf.Duration duration = 2;
 }
 
+// TODO: Remove this event after grpc server is removed.
 message GetEvaluationSizeMetricsEvent {
   map<string, string> labels = 1;
   int32 size_byte = 2;
 }
 
+enum APIId {
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:84]: nit: `ApiId`

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

**Comment by cre8ivejp** [comment]
We discussed adding also `NetworkErrorMetricsEvent` and `InternalSdkErrorMetricsEvent`, but it is missing in the PR.

File: proto/event/client/event.proto [hunk 7]
```diff
@@ -69,24 +69,59 @@ message MetricsEvent {
...
 
+message TimeoutErrorNumberMetricsEvent {
+  APIId api_id = 1;
+  string tag = 2 [deprecated = true];
+  map<string, string> labels = 3;
+}
+
+message InternalErrorNumberMetricsEvent {
+  APIId api_id = 1;
+  string tag = 2 [deprecated = true];
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:121]: Same as [here](https://github.com/bucketeer-io/bucketeer/pull/29/files#r997638101).

**Comment by cre8ivejp** [comment]
Please rename the final commit message when you merge to `chore: add new sdk metrics event types to proto`.
The `feat` is used for a new or a feature update.

File: proto/event/client/event.proto [hunk 8]
```diff
@@ -70,23 +70,56 @@ message MetricsEvent {
...
 }
 
+message TimeoutErrorMetricsEvent {
+  ApiId api_id = 1;
+  map<string, string> labels = 3;
+}
+
+message InternalErrorMetricsEvent {
+  ApiId api_id = 1;
+  map<string, string> labels = 3;
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:120]: Same as above

File: proto/event/client/event.proto [hunk 9]
```diff
@@ -70,23 +70,56 @@ message MetricsEvent {
...
 }
 
 message InternalErrorCountMetricsEvent {
+  option deprecated = true;
   string tag = 1;
 }
 
+message TimeoutErrorMetricsEvent {
+  ApiId api_id = 1;
+  map<string, string> labels = 3;
```
Comment (cre8ivejp) [inline proto/event/client/event.proto:115]: Please change the id to 2.

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

**Review by cre8ivejp — APPROVED** [review]
Awesome!
Thanks!

**Pull request merged**